### ISSUE TITLE
Frontier client for python is not working in python3

### ISSUE THREAD

**Issue by mrodozov** [cms-sw/cmssw#28148]
While trying to make unit tests work in python3 in the PY3 IB, 
packages that depend on CondCore/Utilities/python/conddblib.py
and subsequently on the frontier python client distributed with this external:

https://github.com/cms-sw/cmsdist/blob/IB/CMSSW_11_0_X/master/frontier_client.spec

are failing in here:

https://github.com/mrodozov/cmssw/commit/5d204d2c81b043b6d18b9a862ae395fba5d6dadb#diff-85c0d1a442773ec4f8dcf31682d014f6R63

To reproduce it this changes:
https://github.com/mrodozov/cmssw/commit/5d204d2c81b043b6d18b9a862ae395fba5d6dadb
 needs to be applied on frontier.py & frontier_client.py in frontier_client/${version}/python/lib  and on  sqlalchemy/connectors/frontier.py . We'll update the frontier_client external to be available in PYTHON3PATH in the PY3 IBs (now added by hand)

Once adjusted use 
scram b runtests_test_MpsWorkFlow
to make it fail, although some (most?) of this tests:

https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_amd64_gcc820/CMSSW_11_0_PY3_X_2019-10-09-2300/unitTestLogs/CondTools/Ecal#/

are failing for the same reason

@ggovi in this files (conddblib.py frontier.py etc) was pointed as maintainer
@davidlange6 @smuzaffar 

In short this db stuff needs adjustments.

**Comment by smuzaffar** [comment]
assign db
@ggovi, we have this patch for frontier_client https://github.com/cms-sw/cmsdist/blob/9f5158fc0fb3aef200da6e53db1cfe1cef5abc5e/frontier_client-2.8.20-add-python-dbapi.patch which basically added 4 python dbapi script in frontier client package. I think we should move these to cmssw itself. these scripts are needed by frontier client itself. Can you please suggest a cmssw package where these scripts should land?

**Comment by davidlange6** [comment]
I agree with this.

However - meanwhile, could we patch the cmsdist patches for frontier client in the externals and get a new IB? Its a pain to use the patch suggested in this issue (plus we can easily check that they will work with python2 that way)

**Comment by davidlange6** [comment]
#28157 fixes stuff that i suspect @mrodozov already did locally.
I believe the bigger issue is boost python usage - perhaps just using python3 in the boost build will help. but let me look more

**Comment by davidlange6** [comment]
with my PR and @mrodozov patches to the frontier client patches (fantastic...)

I get 

[2019-10-11 10:09:08,433] INFO: Connecting to pro [frontier://PromptProd/CMS_CONDITIONS]
[2019-10-11 10:09:08,667] ERROR: module 'urllib' has no attribute 'unquote_plus'

which is coming from py2-sqlalchemy/1.3.8/lib/python3.6/site-packages/sqlalchemy/util/compat.py

which means sqlalchemy is getting called thinking its python2.

i would suggest we build boost with python3 and then come back to this.

**Comment by davidlange6** [comment]
yes, if you look at compat.py it is already checking python version and trying to do the right thing. Thats what I'm puzzled by

**Comment by davidlange6** [comment]
so after the PRs of today, the current issue appears to be this one:

[2019-10-11 15:53:47,899] INFO: Connecting to pro [frontier://PromptProd/CMS_CONDITIONS]
[2019-10-11 15:53:47,971] ERROR: (-3, b'[fn-urlparse.c:59]: config error: bad url f')
last run = DEBUG

**Comment by mrodozov** [comment]
I've stuck in the same place. The 'f' is the entire url according to 
https://github.com/fermitools/frontier/blob/master/client/http/fn-urlparse.c#L59

**Comment by davidlange6** [comment]
There are failed unit tests in the pr for Cmssw. Hopefully the two problems are related


On 11 Oct 2019, at 22:09, Mircho Rodozov <notifications@github.com> wrote:

﻿

I've stuck in the same place. The 'f' is the entire url according to
https://github.com/fermitools/frontier/blob/master/client/http/fn-urlparse.c#L59

**Comment by davidlange6** [comment]
hi @mrodozov  @ggovi 

I think i've fixed the cmssw changes w.r.t. enums (and modernized this stuff a bit)

as for the 'f' issue... this is due (at least) to the python->c interface using unicode. I think one just needs to convert the unicode to byte strings in the interface within frontier_client.py - I didn't make it work yet 

eg
buflen = libfc.fn_gzip_str2urlenc(string, len(string), ctypes.byref(buf))
needs to be 
buflen = libfc.fn_gzip_str2urlenc(string.encode('ascii'), len(string), ctypes.byref(buf))

**Comment by mrodozov** [comment]
the tests
CondTools/RunInfo
CondTools/SiStrip
are failing in this urlparse on line 59 when I apply the frontier fixes. I better dig into fixing the encoding stuff.

**Comment by mrodozov** [comment]
Update is:
encodin the url in frontier here:

solves this 
ERROR: (-3, b'[fn-urlparse.c:59]: config error: bad url f')
but then the conddb using internally sqlachemy fails like this:
1. setup in CondCore/Utilities/scripts/conddb the conddb and sqlalchemy debug logging
```
args = parser.parse_args()                                                                                                                                                                                                       
    logging.basicConfig(                                                                                                                                                                                                             
        format = '[%(asctime)s] %(levelname)s: %(message)s',                                                                                                                                                                         
        level = logging.DEBUG #if args.verbose is not None and args.verbose >= 1 else logging.INFO,                                                                                                                                  
    )                                                                                                                                                                                                                                
    logging.getLogger('sqlalchemy.engine').setLevel(logging.DEBUG)
```
2.
in the main function, then run with py2 first:
python2, run:
`python conddb --yes copy runinfo_start_31X_hlt --destdb runinfo_O2O_test.db --o2oTest`
output:
```
[2019-10-17 15:40:54,485] DEBUG: Preparing connection to pro ...
[2019-10-17 15:40:54,563] DEBUG: connection string set to "oracle+frontier://@frontier%3A%2F%2F%28proxyconfigurl%3Dhttp%3A%2F%2Fgrid-wpad%2Fwpad.dat%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbproxy.fnal.gov%3A3128%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier1.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier2.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier3.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier4.cern.ch%3A8000%2FPromptProd%29%2FPromptProd/CMS_CONDITIONS"
[2019-10-17 15:40:54,565] INFO: Connecting to pro [frontier://PromptProd/CMS_CONDITIONS]
[2019-10-17 15:40:54,565] DEBUG: DB url: oracle+frontier://@frontier%3A%2F%2F%28proxyconfigurl%3Dhttp%3A%2F%2Fgrid-wpad%2Fwpad.dat%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbproxy.fnal.gov%3A3128%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier1.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier2.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier3.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier4.cern.ch%3A8000%2FPromptProd%29%2FPromptProd/CMS_CONDITIONS
[2019-10-17 15:40:54,600] DEBUG:  ... using db "oracle+frontier://@frontier%3A%2F%2F%28proxyconfigurl%3Dhttp%3A%2F%2Fgrid-wpad%2Fwpad.dat%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbproxy.fnal.gov%3A3128%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier1.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier2.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier3.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier4.cern.ch%3A8000%2FPromptProd%29%2FPromptProd/CMS_CONDITIONS", schema "CMS_CONDITIONS"
[2019-10-17 15:40:54,600] DEBUG: Loading db types...
[2019-10-17 15:40:54,630] INFO: SELECT CAST('test plain returns' AS VARCHAR(60 CHAR)) AS anon_1 FROM DUAL
[2019-10-17 15:40:54,630] INFO: {}
[2019-10-17 15:40:54,632] INFO: SELECT CAST('test unicode returns' AS VARCHAR2(60 CHAR)) AS anon_1 FROM DUAL
[2019-10-17 15:40:54,633] INFO: {}
[2019-10-17 15:40:54,634] INFO: SELECT CAST('test nvarchar2 returns' AS NVARCHAR2(60)) AS 
```
python3, same thing:
output:
```
[2019-10-17 15:37:30,721] DEBUG: Preparing connection to pro ...
[2019-10-17 15:37:30,804] DEBUG: connection string set to "oracle+frontier://@frontier%3A%2F%2F%28proxyconfigurl%3Dhttp%3A%2F%2Fgrid-wpad%2Fwpad.dat%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbproxy.fnal.gov%3A3128%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier1.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier2.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier3.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier4.cern.ch%3A8000%2FPromptProd%29%2FPromptProd/CMS_CONDITIONS"
[2019-10-17 15:37:30,806] INFO: Connecting to pro [frontier://PromptProd/CMS_CONDITIONS]
[2019-10-17 15:37:30,806] DEBUG: DB url: oracle+frontier://@frontier%3A%2F%2F%28proxyconfigurl%3Dhttp%3A%2F%2Fgrid-wpad%2Fwpad.dat%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbproxy.fnal.gov%3A3128%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier1.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier2.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier3.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier4.cern.ch%3A8000%2FPromptProd%29%2FPromptProd/CMS_CONDITIONS
[2019-10-17 15:37:30,843] DEBUG:  ... using db "oracle+frontier://@frontier%3A%2F%2F%28proxyconfigurl%3Dhttp%3A%2F%2Fgrid-wpad%2Fwpad.dat%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbproxy.fnal.gov%3A3128%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier1.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier2.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier3.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier4.cern.ch%3A8000%2FPromptProd%29%2FPromptProd/CMS_CONDITIONS", schema "CMS_CONDITIONS"
[2019-10-17 15:37:30,843] DEBUG: Loading db types...
[2019-10-17 15:37:30,873] INFO: SELECT CAST('test plain returns' AS VARCHAR(60 CHAR)) AS anon_1 FROM DUAL
[2019-10-17 15:37:30,873] INFO: {}
/build/mrodozov/CondTools/build_py3_tool_conf_with_frontier_changes/slc7_amd64_gcc820/external/py2-sqlalchemy/1.3.8-cms/lib/python3.6/site-packages/sqlalchemy/engine/default.py:356: SAWarning: Exception attempting to detect unicode returns: ProgrammingError("(frontier.ProgrammingError) Error while fetching data: (-8, b'[frontier.c:1124]: No more proxies. Last error was: Request 90 on chan 1 failed at Thu Oct 17 15:37:35 2019: -8 [fn-htclient.c:438]: server error (HTTP/1.1 404 Not Found) proxy=cmsbproxy.fnal.gov[131.225.205.134] server=cmsfrontier4.cern.ch')",)
  "detect unicode returns: %r" % de

```
meaning it's failing when it tries the 
`[2019-10-17 15:40:54,632] INFO: SELECT CAST('test unicode returns' AS VARCHAR2(60 CHAR)) AS anon_1 FROM DUAL`

**Comment by mrodozov** [comment]
Update is: 
I think it's actually error is here: (as it says so)
`[2019-10-17 16:16:50,686] ERROR: '>' not supported between instances of 'NoneType' and 'tuple'`
and the ProgrammingError from frontier is just part of the Exception/Warning, 
and the real problem is with the Enums :/

**Comment by davidlange6** [comment]
you have all of the fixes to the enums in your tests?

**Comment by davidlange6** [comment]
I realize that the error above was fixed in a few places - python2 allows such a comparison but python3 does not.

but can you paste in your full output?

**Comment by mrodozov** [comment]
```
conddb --yes copy runinfo_start_31X_hlt --destdb runinfo_O2O_test.db --o2oTest
DEBUG: Preparing connection to pro ...
[2019-10-17 17:04:10,519] DEBUG: connection string set to "oracle+frontier://@frontier%3A%2F%2F%28proxyconfigurl%3Dhttp%3A%2F%2Fgrid-wpad%2Fwpad.dat%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbproxy.fnal.gov%3A3128%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier1.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier2.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier3.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier4.cern.ch%3A8000%2FPromptProd%29%2FPromptProd/CMS_CONDITIONS"
[2019-10-17 17:04:10,521] INFO: Connecting to pro [frontier://PromptProd/CMS_CONDITIONS]
[2019-10-17 17:04:10,521] DEBUG: DB url: oracle+frontier://@frontier%3A%2F%2F%28proxyconfigurl%3Dhttp%3A%2F%2Fgrid-wpad%2Fwpad.dat%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbproxy.fnal.gov%3A3128%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier1.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier2.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier3.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier4.cern.ch%3A8000%2FPromptProd%29%2FPromptProd/CMS_CONDITIONS
[2019-10-17 17:04:10,556] DEBUG:  ... using db "oracle+frontier://@frontier%3A%2F%2F%28proxyconfigurl%3Dhttp%3A%2F%2Fgrid-wpad%2Fwpad.dat%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmst0frontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier1.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbpfrontier2.cern.ch%3A3128%29%28backupproxyurl%3Dhttp%3A%2F%2Fcmsbproxy.fnal.gov%3A3128%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier1.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier2.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier3.cern.ch%3A8000%2FPromptProd%29%28serverurl%3Dhttp%3A%2F%2Fcmsfrontier4.cern.ch%3A8000%2FPromptProd%29%2FPromptProd/CMS_CONDITIONS", schema "CMS_CONDITIONS"
[2019-10-17 17:04:10,556] DEBUG: Loading db types...
/build/mrodozov/CondTools/build_py3_tool_conf_with_frontier_changes/slc7_amd64_gcc820/external/py2-sqlalchemy/1.3.8-cms/lib/python3.6/site-packages/sqlalchemy/engine/default.py:356: SAWarning: Exception attempting to detect unicode returns: ProgrammingError("(frontier.ProgrammingError) Error while fetching data: (-8, b'[frontier.c:1124]: No more proxies. Last error was: Request 90 on chan 1 failed at Thu Oct 17 17:04:14 2019: -8 [fn-htclient.c:438]: server error (HTTP/1.1 404 Not Found) proxy=cmsbproxy.fnal.gov[131.225.205.134] server=cmsfrontier4.cern.ch')",)
  "detect unicode returns: %r" % de
/build/mrodozov/CondTools/build_py3_tool_conf_with_frontier_changes/slc7_amd64_gcc820/external/py2-sqlalchemy/1.3.8-cms/lib/python3.6/site-packages/sqlalchemy/engine/default.py:356: SAWarning: Exception attempting to detect unicode returns: ProgrammingError("(frontier.ProgrammingError) Error while fetching data: (-8, b'[frontier.c:1124]: No more proxies. Last error was: Request 180 on chan 1 failed at Thu Oct 17 17:04:19 2019: -8 [fn-htclient.c:438]: server error (HTTP/1.1 404 Not Found) proxy=cmsbproxy.fnal.gov[131.225.205.134] server=cmsfrontier4.cern.ch')",)
  "detect unicode returns: %r" % de
/build/mrodozov/CondTools/build_py3_tool_conf_with_frontier_changes/slc7_amd64_gcc820/external/py2-sqlalchemy/1.3.8-cms/lib/python3.6/site-packages/sqlalchemy/engine/default.py:356: SAWarning: Exception attempting to detect unicode returns: ProgrammingError("(frontier.ProgrammingError) Error while fetching data: (-8, b'[frontier.c:1124]: No more proxies. Last error was: Request 270 on chan 1 failed at Thu Oct 17 17:04:22 2019: -8 [fn-htclient.c:438]: server error (HTTP/1.1 404 Not Found) proxy=cmsbproxy.fnal.gov[131.225.205.134] server=cmsfrontier4.cern.ch')",)
  "detect unicode returns: %r" % de
[2019-10-17 17:04:22,515] ERROR: '>' not supported between instances of 'NoneType' and 'tuple'
```
the comparison comes from conddb , not from sqlalchemy (I think)

**Comment by smuzaffar** [comment]
this has been fixed
